Artifacts/MsSql/Structure/Security/0040-Tables - RLPhase2.sql

```diff
@@ -0,0 +1,132 @@
+REFERENCES [dbo].[AuthorizationStrategies] ([AuthorizationStrategyId])
+ON DELETE CASCADE
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides] CHECK CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides_dbo.AuthorizationStrategies_AuthorizationStrategy_Authorization]
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides_dbo.ClaimSetResourceClaimActionAuthorizations_ClaimSetResourceC] FOREIGN KEY([ClaimSetResourceClaimActionAuthorization_ClaimSetResourceClaimActionAuthorizationId])
+REFERENCES [dbo].[ClaimSetResourceClaimActionAuthorizations] ([ClaimSetResourceClaimActionAuthorizationId])
+GO
+
+ALTER TABLE [dbo].[ClaimSetResourceClaimActionAuthorizationStrategyOverrides] CHECK CONSTRAINT [FK_dbo.ClaimSetResourceClaimActionAuthorizationStrategyOverrides_dbo.ClaimSetResourceClaimActionAuthorizations_ClaimSetResourceC]
+GO
+
+
+------------------------------ [dbo].[ResourceClaimActionAuthorizations] -------------------------------------------------
+CREATE TABLE [dbo].[ResourceClaimActionAuthorizations](
+	[ResourceClaimActionAuthorizationId] [int] IDENTITY(1,1) NOT NULL,
+	[ValidationRuleSetName] [nvarchar](255) NULL,
+	[Action_ActionId] [int] NOT NULL,
+	[ResourceClaim_ResourceClaimId] [int] NOT NULL,
+ CONSTRAINT [PK_dbo.ResourceClaimActionAuthorizations] PRIMARY KEY CLUSTERED 
+(
+	[ResourceClaimActionAuthorizationId] ASC
+)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
+) ON [PRIMARY]
+GO
+
+ALTER TABLE [dbo].[ResourceClaimActionAuthorizations]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ResourceClaimActionAuthorizations_dbo.Actions_Action_ActionId] FOREIGN KEY([Action_ActionId])
+REFERENCES [dbo].[Actions] ([ActionId])
+ON DELETE CASCADE
+GO
+
+ALTER TABLE [dbo].[ResourceClaimActionAuthorizations] CHECK CONSTRAINT [FK_dbo.ResourceClaimActionAuthorizations_dbo.Actions_Action_ActionId]
+GO
+
+ALTER TABLE [dbo].[ResourceClaimActionAuthorizations]  WITH CHECK ADD  CONSTRAINT [FK_dbo.ResourceClaimActionAuthorizations_dbo.ResourceClaims_ResourceClaim_ResourceClaimId] FOREIGN KEY([ResourceClaim_ResourceClaimId])
+REFERENCES [dbo].[ResourceClaims] ([ResourceClaimId])
+ON DELETE CASCADE
+GO
+
+ALTER TABLE [dbo].[ResourceClaimActionAuthorizations] CHECK CONSTRAINT [FK_dbo.ResourceClaimActionAuthorizations_dbo.ResourceClaims_ResourceClaim_ResourceClaimId]
+GO
+
+------------------------------ [dbo].[ResourceClaimActionAuthorizationStrategies] -------------------------------------------------
+
+
+CREATE TABLE [dbo].[ResourceClaimActionAuthorizationStrategies](
```

Comment:
Here we should add a unique constraint for `ResourceClaimActionAuthorization_ResourceClaimActionAuthorizationId` and `AuthorizationStrategy_AuthorizationStrategyId` as it should not be possible to add the same `AuthorizationStrategyId` twice for the same resource claim action authorization.

@vimayya Thoughts?